tensorboard/plugins/pr_curve/summary.py

```diff
@@ -31,24 +31,19 @@ def op(
     labels,
     predictions,
     num_thresholds=200,
-    weights=None,
-    num_classes=1,
+    weight=1.0,
     display_name=None,
-    description=None):
-  """Computes multi-class PR summaries for a list of thresholds in `[0, 1]`.
+    description=''):
```

Comment:
Three off-topic comments:
  * IMO, Java serialization is worse than reflection. Reflection is very explicit about what it does, and can be controlled at least to some degree by a `SecurityManager`. Serialization _creates objects without invoking constructors_, which violates a core invariant…so forces you to implement these special `readObject` methods…which force you to initialize fields outside of constructors…so you can't mark your fields `final` even when they actually should be, resulting in a (private) API change…<sub>unless you use reflection…</sub>
  * IMO, Python's `f.func_closure` is far worse than `f.func_defaults`.
  * The good news about `f.func_closure` is that I tried very hard to find a way to mutate those values, and have a sketch of a proof that—at least in CPython—it is not possible without resorting to extension modules.